cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaSpecChecker.java

```diff
@@ -182,4 +187,53 @@ private void checkKRaftControllerCount(List<Condition> warnings)    {
                     "Running KRaft controller quorum with an odd number of nodes is recommended."));
         }
     }
+
+    /**
+     * Checks if the version of the Kafka brokers matches any custom metadata version config. Updating this is the final
+     * step in upgrading Kafka version in KRaft, so if this doesn't match it is possibly an indication that a user has
+     * updated their Kafka cluster and is unaware that they also should update their metadata version to match.
+     *
+     * @param warnings List to add a warning to, if appropriate.
+     */
+    private void checkKafkaMetadataVersion(List<Condition> warnings) {
+        String metadataVersion = kafkaCluster.getMetadataVersion();
+
+        if (metadataVersion != null) {
+            Matcher m = MAJOR_MINOR_REGEX.matcher(metadataVersion);
+            if (m.matches() && !kafkaBrokerVersion.startsWith(m.group(1))) {
+                warnings.add(StatusUtils.buildWarningCondition("KafkaMetadataVersion",
+                        "Metadata version is older than the Kafka version used by the cluster, which suggests that an upgrade is incomplete."));
+            }
+        }
+    }
+
+    /**
+     * inter.broker.protocol.version should not be used in KRaft. This method checks if it is set and in case it is, it
+     * raises the warning.
+     *
+     * @param warnings List to add a warning to, if appropriate.
+     */
+    private void checkInterBrokerProtocolVersionInKRaft(List<Condition> warnings) {
+        String interBrokerProtocolVersion = kafkaCluster.getConfiguration().getConfigOption(KafkaConfiguration.INTERBROKER_PROTOCOL_VERSION);
+
+        if (interBrokerProtocolVersion != null) {
+            warnings.add(StatusUtils.buildWarningCondition("KafkaInterBrokerProtocolVersionInKRaft",
+                    "inter.broker.protocol.version is not used in KRaft-based Kafka clusters and should be removed from the Kafka custom resource."));
+        }
```

Comment:
We should be aware that this warning will be in the Kafka status during a migration, because IBP is still set on brokers and the cluster is also using KRaft at the same time.
Of course, it would be transient.

Reply:
Why would it be there? It is gated by `useKRaft`, so it should be called only in KRaft mode.

Reply:
It's not actually true on my migration work. It's because the `useKRaft` passed to the checker is defined like this:

```java
result.warningConditions.addAll(specChecker.run(kafkaMetadataConfigState.isPreMigrationOrKRaft()));
```

So it's true since PRE_MIGRATION phase when the KRaft controllers are deployed, not just at the end when in full KRAFT state. It's because when deploying the controllers, we want the checker validating the first two things: controller storage and controller count. So I had to "adapt" the flag that way.
Anyway I don't think having the warning is a big issue, the opposite I think it's important so the user knows to remove IBP and LMF at the end of migration (or anyway the warning will disappear if the operator added and then removed them because not specified by user).

If we want to avoid that warning earlier, I should change in my branch the signature of the `run` method getting the configuration state and not just a boolean and then running the controller storage and count check if in PRE_MIGRATION while all the others only if in KRAFT.

Reply:
I would probably leave the warnings for KRaft mode only. It is just warnings anyway, so if you get them only after the migration is done, that is fine.

Reply:
You could also pass the `kafkaMetadataConfigState` into the spec checker and split the warnings. Do thse from PreMigration:
```
checkKRaftControllerStorage(warnings);
checkKRaftControllerCount(warnings);
checkKafkaMetadataVersion(warnings);
```

and these only in KRaft:

```
checkInterBrokerProtocolVersionInKRaft(warnings);
checkLogMessageFormatVersionInKRaft(warnings);
```

Reply:
Ah ah ah ... we are overlapping the discussion :-)
I am fine with both approaches. As I said in the end we warn but don't stop the migration anyway (about controllers count and storage).

Reply:
Yeah, let's leave the check for the end of the migration. Anyway, again we just warn but not stopping the migration.